src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,125 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class Storage {
+    private static final String ROOT_PROJECT = System.getProperty("user.dir");
+    private static Path saveFilePath =
+            Paths.get(ROOT_PROJECT, "src", "data", "duke.txt");
+    private static Path saveFolderPath =
+            Paths.get(ROOT_PROJECT, "src", "data");
+
+    /**
+     * Constructs a Storage object.
+     * @param filePath Path of file
+     * @throws DukeException If database cannot be created or found.
+     */
+    public Storage(Path filePath) throws DukeException {
+        createDbIfNotFound();
+    }
+
+    /**
+     * Returns the path for the address to save the file.
+     * @return Path
+     */
+    public static Path getSaveFilePath() {
+        return saveFilePath;
+    }
+
+    /**
+     * Creates a file w/wo directory for storing of file.
+     * @throws DukeException If access rights issue is violated.
+     */
+    protected static void createDbIfNotFound() throws DukeException {
+        try {
+            if (Files.notExists(saveFilePath)) {
+                if (Files.notExists(saveFolderPath)) {
+                    saveFolderPath = Files.createDirectories(saveFolderPath);
+                }
+                saveFilePath = Files.createFile(saveFilePath);
+            }
+        } catch (IOException e) {
+            throw new DukeException("Unable to access data stored."
+                    + e);
```

Comment:
is break really necessary here since it only contains one `e`? keeping the throw statement one line can make it look more concise?